util/write_batch_util.h

```diff
@@ -0,0 +1,81 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+#include <unordered_set>
+#include <vector>
+
+#include "rocksdb/slice.h"
+#include "rocksdb/status.h"
+#include "rocksdb/write_batch.h"
+
+namespace ROCKSDB_NAMESPACE {
+// ColumnFamilyCollector is a write batch handler which does nothing
+// except recording unique column family IDs
+class ColumnFamilyCollector : public WriteBatch::Handler {
+  std::unordered_set<uint32_t> column_family_ids_;
+
+  Status AddColumnFamilyId(uint32_t column_family_id) {
+    if (column_family_ids_.find(column_family_id) == column_family_ids_.end()) {
```

Comment:
Minor but this `find` is redundant; just calling `insert` unconditionally yields the same result with fewer lookups